File: exercises/java/src/test/java/checkout/CheckoutTests.java
package checkout;

import org.junit.Test;

import static org.assertj.core.api.Assertions.*;

public class CheckoutTests {

    @Test
    public void oneA() {
        Checkout checkout = new Checkout();

        checkout.scan("A");
        assertThat(checkout.total()).isEqualTo(50);
    }

    @Test
    public void twoA() {
        Checkout checkout = new Checkout();

        checkout.scan("A");
        checkout.scan("A");
        assertThat(checkout.total()).isEqualTo(100);
    }

    @Test
    public void threeA() {
        Checkout checkout = new Checkout();

        checkout.scan("A");
        checkout.scan("A");
        checkout.scan("A");
        assertThat(checkout.total()).isEqualTo(130);
    }


    @Test
    public void sixA() {
        Checkout checkout = new Checkout();

        checkout.scan("A");
        checkout.scan("A");
        checkout.scan("A");
        checkout.scan("A");
        checkout.scan("A");
        checkout.scan("A");
        assertThat(checkout.total()).isEqualTo(260);
    }

    @Test
    public void oneB() {
        Checkout checkout = new Checkout();

        checkout.scan("B");
        assertThat(checkout.total()).isEqualTo(30);
    }

    @Test
    public void twoB() {
        Checkout checkout = new Checkout();

        checkout.scan("B");
        checkout.scan("B");
        assertThat(checkout.total()).isEqualTo(45);
    }

    @Test
    public void fourB() {
        Checkout checkout = new Checkout();

        checkout.scan("B");
        checkout.scan("B");
        checkout.scan("B");
        checkout.scan("B");
        assertThat(checkout.total()).isEqualTo(90);
    }

    @Test
    public void simple() {
        Checkout checkout = new Checkout();
        
        checkout.scan("A");
        assertThat(checkout.total()).isEqualTo( 50);
        
        checkout.scan("B");
        assertThat(checkout.total()).isEqualTo( 80);
        
        checkout.scan("C");
        assertThat(checkout.total()).isEqualTo(100);
        
        checkout.scan("D");
        assertThat(checkout.total()).isEqualTo(115);
    }
}

File: exercises/dotnet/McrDigital.Bootcamp1.Checkout.Tests/CheckoutTests.cs
namespace McrDigital.Bootcamp1.Checkout.Tests
{
    using Xunit;
    using McrDigital.Bootcamp1.Checkout;

    public class CheckoutTests
    {
        [Fact]
        public void OneA()
        {
            var checkout = new Checkout();

            checkout.Scan("A");

            Assert.Equal(50, checkout.Total);
        }

        [Fact]
        public void TwoA()
        {
            var checkout = new Checkout();

            checkout.Scan("A");
            checkout.Scan("A");

            Assert.Equal(100, checkout.Total);
        }

        [Fact]
        public void ThreeA()
        {
            var checkout = new Checkout();

            checkout.Scan("A");
            checkout.Scan("A");
            checkout.Scan("A");

            Assert.Equal(130, checkout.Total);
        }

        [Fact]
        public void SixA()
        {
            var checkout = new Checkout();

            checkout.Scan("A");
            checkout.Scan("A");
            checkout.Scan("A");
            checkout.Scan("A");
            checkout.Scan("A");
            checkout.Scan("A");

            Assert.Equal(260, checkout.Total);
        }

        [Fact]
        public void OneB()
        {
            var checkout = new Checkout();

            checkout.Scan("B");

            Assert.Equal(30, checkout.Total);
        }

        [Fact]
        public void TwoB()
        {
            var checkout = new Checkout();

            checkout.Scan("B");
            checkout.Scan("B");

            Assert.Equal(45, checkout.Total);
        }

        [Fact]
        public void FourB()
        {
            var checkout = new Checkout();

            checkout.Scan("B");
            checkout.Scan("B");
            checkout.Scan("B");
            checkout.Scan("B");

            Assert.Equal(90, checkout.Total);
        }

        [Fact]
        public void Simple()
        {
            var checkout = new Checkout();

            checkout.Scan("A");
            Assert.Equal(50, checkout.Total);

            checkout.Scan("B");
            Assert.Equal(80, checkout.Total);

            checkout.Scan("C");
            Assert.Equal(100, checkout.Total);

            checkout.Scan("D");
            Assert.Equal(115, checkout.Total);
        }

        [Fact]
        public void Incremental()
        {
            var checkout = new Checkout();

            checkout.Scan("A");
            Assert.Equal(50, checkout.Total);

            checkout.Scan("B");
            Assert.Equal(80, checkout.Total);

            checkout.Scan("A");
            Assert.Equal(130, checkout.Total);

            checkout.Scan("A");
            Assert.Equal(160, checkout.Total);

            checkout.Scan("B");
            Assert.Equal(175, checkout.Total);

            checkout.Scan("C");
            Assert.Equal(195, checkout.Total);

            checkout.Scan("B");
            Assert.Equal(225, checkout.Total);

            checkout.Scan("C");
            Assert.Equal(245, checkout.Total);

            checkout.Scan("D");
            Assert.Equal(260, checkout.Total);

            checkout.Scan("D");
            Assert.Equal(275, checkout.Total);

            checkout.Scan("D");
            Assert.Equal(290, checkout.Total);

            checkout.Scan("C");
            Assert.Equal(310, checkout.Total);
        }
    }
}